### ISSUE TITLE
Recovery fails for InmemJournal when events have been tagged

### ISSUE THREAD

**Issue by odd** [akka/akka#28552]
Tagged events are not able to be recovered when using `akka.persistence.journal.inmem.InmemJournal`; instead a class cast exception is thrown. 

```scala
import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
import akka.actor.typed.{ActorRef, Behavior}
import akka.persistence.typed.scaladsl.{Effect, EventSourcedBehavior}
import akka.persistence.typed.PersistenceId
import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, Matchers, WordSpecLike}

object Adder {
  sealed trait Command
  case class Add(n: Int, replyTo: ActorRef[Int]) extends Command
  case object Passivate extends Command
  sealed trait Event
  case class Added(n: Int) extends Event

  def apply(initialValue: Int): Behavior[Command] =
    EventSourcedBehavior[Command, Event, Int](
      persistenceId = PersistenceId(Adder.getClass.getName, initialValue.toString),
      emptyState = initialValue,
      commandHandler = (_: Int, cmd: Command) => {
        cmd match {
          case Add(n, replyTo) =>
            Effect.persist(Added(n)).thenReply(replyTo)(total => total)
          case Passivate =>
            Effect.none.thenStop()
        }
      },
      eventHandler = (total: Int, evt: Event) => {
        evt match {
          case Added(n) => total + n
        }
      }).withTagger(_ => Set(s"adding"))
}
class InmemTaggingSpec
    extends ScalaTestWithActorTestKit
    with WordSpecLike
    with BeforeAndAfter
    with BeforeAndAfterAll
    with Matchers {

  "A tagged event" must {
    "be replayable" in {
      val adder1 = testKit.spawn(Adder(5))
      val probe = testKit.createTestProbe[Int]()
      adder1 ! Adder.Add(3, probe.ref)
      probe.expectMessage(8)
      adder1 ! Adder.Passivate
      val adder2 = testKit.spawn(Adder(5))
      adder2 ! Adder.Add(3, probe.ref)
      probe.expectMessage(11)
    }
  }
}
``` 
The test case above results in: `java.lang.ClassCastException: class akka.persistence.journal.Tagged cannot be cast to class Adder$Event` (see full stacktrace below).

```
akka.persistence.typed.internal.JournalFailureException: Exception during recovery. Last known sequence number [1]. PersistenceId [Adder$|5], due to: Exception during recovery while handling [akka.persistence.journal.Tagged] with sequence number [1]. PersistenceId [Adder$|5], due to: class akka.persistence.journal.Tagged cannot be cast to class Adder$Event (akka.persistence.journal.Tagged and Adder$Event are in unnamed module of loader 'app')
	at akka.persistence.typed.internal.ReplayingEvents.onRecoveryFailure(ReplayingEvents.scala:221)
	at akka.persistence.typed.internal.ReplayingEvents.onJournalResponse(ReplayingEvents.scala:153)
	at akka.persistence.typed.internal.ReplayingEvents.onMessage(ReplayingEvents.scala:90)
	at akka.persistence.typed.internal.ReplayingEvents.onMessage(ReplayingEvents.scala:65)
	at akka.actor.typed.scaladsl.AbstractBehavior.receive(AbstractBehavior.scala:82)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
	at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:55)
	at akka.persistence.typed.internal.EventSourcedBehaviorImpl$$anon$1.aroundReceive(EventSourcedBehaviorImpl.scala:157)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:83)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
	at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:55)
	at akka.actor.typed.internal.SimpleSupervisor.aroundReceive(Supervision.scala:123)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:83)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
	at akka.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:126)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:106)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
	at akka.actor.ActorCell.invoke(ActorCell.scala:543)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: akka.persistence.typed.internal.JournalFailureException: Exception during recovery while handling [akka.persistence.journal.Tagged] with sequence number [1]. PersistenceId [Adder$|5], due to: class akka.persistence.journal.Tagged cannot be cast to class Adder$Event (akka.persistence.journal.Tagged and Adder$Event are in unnamed module of loader 'app')
	at akka.persistence.typed.internal.ReplayingEvents.onRecoveryFailure(ReplayingEvents.scala:221)
	at akka.persistence.typed.internal.ReplayingEvents.onJournalResponse(ReplayingEvents.scala:133)
	... 27 common frames omitted
Caused by: java.lang.ClassCastException: class akka.persistence.journal.Tagged cannot be cast to class Adder$Event (akka.persistence.journal.Tagged and Adder$Event are in unnamed module of loader 'app')
	at akka.persistence.typed.internal.ReplayingEvents.handleEvent$1(ReplayingEvents.scala:119)
	at akka.persistence.typed.internal.ReplayingEvents.onJournalResponse(ReplayingEvents.scala:124)
	... 27 common frames omitted
```

**Comment by manonthegithub** [comment]
Does the fix in #28553 support persistent queries? Maybe I am wrong, but Tags are not saved and therefore they are lost.

**Comment by patriknw** [comment]
The InmemJournal doesn't support queries, but it should still be possible to persist/replay events wrapped in Tagged.
